it/uk/gov/hmrc/agentstatuschange/support/UpstreamServicesStubs.scala

```diff
@@ -0,0 +1,66 @@
+package uk.gov.hmrc.agentstatuschange.support
+
+import java.net.URL
+
+import com.github.tomakehurst.wiremock.WireMockServer
+import com.github.tomakehurst.wiremock.client.WireMock._
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration._
+import org.scalatest.concurrent.Eventually
+import org.scalatest.{BeforeAndAfterAll, BeforeAndAfterEach, Suite}
+import play.api.libs.json.Json
+import uk.gov.hmrc.play.it.Port
+
+trait UpstreamServicesStubs extends BeforeAndAfterAll
+  with BeforeAndAfterEach with Eventually {
+
+  this: Suite =>
+
+  val wireMockHost = "localhost"
+  val wireMockPort: Int = Port.randomAvailable
+  val wireMockBaseUrlAsString = s"http://$wireMockHost:$wireMockPort"
+  lazy val wireMockServer = new WireMockServer(wireMockConfig().port(wireMockPort))
+
+  val fakeCredId = "fakeCredId"
+  val fakeCredIdType = "GovernmentGateway"
+
+  override protected def beforeAll(): Unit = {
+    super.beforeAll()
+    configureFor(wireMockHost, wireMockPort)
+    wireMockServer.start()
+  }
+
+  override protected def afterAll(): Unit = {
+    wireMockServer.stop()
+    super.afterAll()
+  }
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+    reset()
+    givenAuthReturnsUserDetails()
+    givenImplicitAuditEvents()
+  }
+
+  def givenImplicitAuditEvents() = {
+    stubFor(post(urlPathEqualTo("/write/audit/merged")).willReturn(aResponse().withStatus(204)))
```

Comment:
Not sure if we'd need a stub here for datastream's "/write/audit" endpoint as well. If there's wiremock warnings in the logs whilst testing, we could add another line here like:
```
stubFor(post(urlPathEqualTo("/write/audit")).willReturn(aResponse().withStatus(204)))
```

Reply:
👍